contrib/xor-blocks/src/main.rs

```diff
@@ -0,0 +1,130 @@
+use std::{
+    env, fs,
+    io::{self, Read},
+    path::Path,
+    time::{Duration, Instant},
+};
+
+const MAGIC: [u8; 4] = [0xf9, 0xbe, 0xb4, 0xd9];
+
+fn main() -> Result<(), io::Error> {
+    let start = Instant::now();
+
+    let blocks_path = if env::args().len() > 1 {
+        env::args().nth(1).expect("the arg to exist").into()
+    } else {
+        #[allow(deprecated)]
+        env::home_dir()
+            .expect("to have a home directory")
+            .join(match env::consts::OS {
+                "macos" => "Library/Application Support/Bitcoin",
+                "windows" => "AppData\\Local\\Bitcoin",
+                "linux" => ".bitcoin",
+                _ => {
+                    println!("Unknown OS");
+                    return Ok(());
```

Comment:
This is an error and should return non-zero exit code.

Reply:
Done.